pkgs/dart_services/lib/src/sdk.dart

```diff
@@ -46,15 +46,23 @@ class Sdk {
     return const [];
   }
 
-  // When running with `dart run`, the FLUTTER_ROOT environment variable is
-  // set automatically.
   String _getSdkPath() {
-    final env = Platform.environment;
-    if (!env.containsKey('FLUTTER_ROOT') || env['FLUTTER_ROOT']!.isEmpty) {
-      throw Exception('No FLUTTER_ROOT variable set');
+    // <flutter-sdk>/bin/cache/dart-sdk
+    final dart = Platform.resolvedExecutable;
+    final dartSdk = path.dirname(path.dirname(dart));
+    final flutterSdk = path.dirname(path.dirname(path.dirname(dartSdk)));
```

Comment:
(nit) We might want to consider making `sdkPath` a field on the `Sdk` class. Right now we are caching the value globally, and then the `Sdk` constructor uses this value to get the Dart executable path back out again.

Reply:
👍 - I took a stab at that in this PR